### ISSUE TITLE
Yalc link doesn't update anything when modifying the linked package

### ISSUE THREAD

**Issue by nialna** [whitecolor/yalc#116]
I'm not sure I understand how yalc link is meant to be used. I tried setting it up to be able to have quick editing of packages, so:

* I have a watch command running in my package which rebuilds it
* The project using the package via yalc has it setup with `yalc link`
* When the package is rebuilt, nothing is updated in the local version of the project
* If I try to manually use `yalc publish` in the package, it still doesn't update in my project
* If I also use `yalc push` to make it update my project it does appear

So I don't get what `yalc link` is actually meant for if the link doesn't get updated when changing local package files, and still doesn't get updated when publishing? I still have to use `yalc push` for the files to change in my project, so I'm not sure if I misunderstood something

**Comment by isokosan** [comment]
This is the exact situation I found myself in. I used inotifywatch to setup a watcher and run yalc push on file change events following the StackOverflow answer here:
[https://stackoverflow.com/a/50115609](https://stackoverflow.com/a/50115609)

Modified it as below:
```
#!/bin/bash
while true; do

inotifywait -e modify,create,delete,move -r $1 && \
yalc push --private

done
```
There is a caveat with having too many files to watch though as described below so you can exclude unnecessary folders.
> Warning: If you use this option while watching the root directory of a large tree, it may take quite a while until all inotify watches are established, and events will not be received in this time. Also, since one inotify watch will be established per subdirectory, it is possible that the maximum amount of inotify watches per user will be reached. The default maximum is 8192; it can be increased by writing to /proc/sys/fs/inotify/max_user_watches.

**Comment by dcorliss-Novetta** [comment]
I'm having the same troubles, except even yalc push doesn't update the linked package in my project. I have to run 'yalc link m-package' in my project every time

**Comment by Flashfyre** [comment]
I have this issue too, where I have to manually run 'yalc publish --push' on a local dependency in order to update it on my dependent. It would be nice if you could trigger this on all changed linked dependencies from the dependent with a single command. This way, it could be added as a script in package.json, which I can't do right now without hardcoding my local directories and first changing to those and running the command on each.

**Comment by lolmaus** [comment]
Yalc needs native support for this! Or at least an officially recommended, well-documented way to leverage third-party tools like `inotify-tools`.

**Comment by wclr** [comment]
> I still have to use yalc push for the files to change in my project, so I'm not sure if I misunderstood something

Yes. Yalc link just allow not to touch `package.json` but it in any case brings code locally to the project.

**Comment by ericcirone** [comment]
So i just found this package, and for installing dependancies of a local dev package this works great with `yalc add`, but yes it loses the auto update when dev code is updated in the _yalced_ package. So to get that functionality back i still use `yarn link`

1. `yalc publish` my dev package
2. `yarn link` my dev package
3. `yalc add` my dev package to my project
4. `yarn link dev-package` in my project

This seems to get me the best of both worlds.

**Comment by Nantris** [comment]
Unfortunately mixing `yalc/yarn` as advised by @ericcirone just breaks our build tools the same as `yarn link` alone does.
